include/open62541pp/types/Variant.h

```diff
@@ -72,54 +69,35 @@ class Variant : public TypeWrapper<UA_Variant, UA_TYPES_VARIANT> {
         return variant;
     }
 
-    /// Create Variant from array (no copy if assignable without conversion).
-    template <typename T>
-    [[nodiscard]] static Variant fromArray(Span<T> array) {
+    /// Create Variant from array (no copy if assignable).
+    template <typename ArrayLike>
+    [[nodiscard]] static Variant fromArray(ArrayLike&& array) {
+        using ValueType = typename std::remove_reference_t<ArrayLike>::value_type;
+        constexpr bool isMutable = detail::IsMutableContainer<ArrayLike>::value;
+        constexpr bool isContiguous = detail::IsContiguousContainer<ArrayLike>::value;
         Variant variant;
-        if constexpr (detail::isRegisteredType<T>) {
+        if constexpr (isMutable && isContiguous && detail::isRegisteredType<ValueType>) {
             variant.setArray(array);
         } else {
             variant.setArrayCopy(array);
         }
         return variant;
     }
 
-    /// Create Variant from array with custom data type (no copy).
-    template <typename T>
-    [[nodiscard]] static Variant fromArray(Span<T> array, const UA_DataType& dataType) {
-        Variant variant;
-        variant.setArray(array, dataType);
-        return variant;
-    }
-
-    /// Create Variant from array (copy).
-    template <typename T>
-    [[nodiscard]] static Variant fromArray(Span<const T> array) {
-        Variant variant;
-        variant.setArrayCopy(array);
-        return variant;
-    }
-
-    /// Create Variant from array with custom data type (copy).
-    template <typename T>
-    [[nodiscard]] static Variant fromArray(Span<const T> array, const UA_DataType& dataType) {
+    /// Create Variant from array with custom data type (no copy if assignable).
+    template <typename ArrayLike>
+    [[nodiscard]] static Variant fromArray(ArrayLike&& array, const UA_DataType& dataType) {
+        constexpr bool isMutable = detail::IsMutableContainer<ArrayLike>::value;
+        constexpr bool isContiguous = detail::IsContiguousContainer<ArrayLike>::value;
         Variant variant;
-        variant.setArrayCopy(array, dataType);
+        if constexpr (isMutable && isContiguous) {
+            variant.setArray(array, dataType);
```

Comment:
forward array?

Reply:
👍 Fixed in https://github.com/open62541pp/open62541pp/pull/174/commits/50f3616665e2015dd8e71b53468491cf887bb58a